File: 2028 find-missing-observations/solution.java
class Solution {
    public int[] missingRolls(int[] rolls, int mean, int n) {
        int m = rolls.length;
        int sum = 0;
        for(int i:rolls) sum+=i;
        int an = mean*(m+n)-sum;
        if(an < n || an > n*6) return new int[0];
        int[] ans = new int[n];
        int b=an/n,r=an%n;
        for(int i=0;i<n;i++) {
            ans[i]=b+( i<r ? 1 : 0);
            if (ans[i] < 1 || ans[i] > 6) return new int[0];
        }
        return ans;
    }
}

File: 2028 find-missing-observations/solution.go
func missingRolls(rolls []int, mean int, n int) []int {
    m:=len(rolls)
    sum:=0
    for _,v:=range rolls {
        sum+=v
    }
    an:=mean*(m+n)-sum;
    if an < n || an > n*6 {
        return []int{}
    }
    ans:=make([]int,n)
    b,r:=an/n,an%n;
    for i:=0;i<n;i++ {
        if i<r {
            ans[i] = b+1;
        } else {
            ans[i] = b
        }
        if ans[i] < 1 || ans[i] > 6 {
            return []int{}
        }
    }
        return ans;
}
